core/templates/services/context.service.ts

```diff
@@ -32,14 +32,63 @@ export class ContextService {
   constructor(
     private urlService: UrlService) {}
 
-  pageContext = null;
-  explorationIsLinkedToStory = false;
-  explorationId = null;
-  questionPlayerIsManuallySet = false;
-  questionId = null;
-  editorContext = null;
-  customEntityContext = null;
-  imageSaveDestination: string = AppConstants.IMAGE_SAVE_DESTINATION_SERVER;
+  static _pageContext = null;
```

Comment:
nit: Let's move all the variable declarations to the top and have the setters and getters below.